src/js/node/events.js

```diff
@@ -92,9 +92,28 @@ const emitWithoutRejectionCapture = function emit(type, ...args) {
   if (events === undefined) return false;
   var handlers = events[type];
   if (handlers === undefined) return false;
-
-  for (var handler of [...handlers]) {
-    handler.apply(this, args);
+  // clone handlers array if necessary since handlers can be added/removed during the loop
+  const maybeClonedHandlers = handlers.length > 1 ? handlers.slice() : handlers;
+  for (let i = 0; i < maybeClonedHandlers.length; i++) {
```

Comment:
```suggestion
  for (let i = 0, { length } = maybeClonedHandlers; i < length; i++) {
```
Engines can inline this access but we can help them out. It's low level but a good habit to be part of the repeated low-level loop scaffolding